Articles/Blog/TheGreatMigration.md

```diff
@@ -0,0 +1,107 @@
+# The Great Migration
+## Experiences migrating 2 million lines of code and 20 years of development history from Subversion to GitHub.
+
+# DRAFT DRAFT DRAFT
+
+**Hero Image:**
+
+- <a href='https://raw.githubusercontent.com/betterscientificsoftware/images/blog_svn_gh_migration/Blog_TheGreatMigration_car.jpg'><img src='https://raw.githubusercontent.com/betterscientificsoftware/images/blog_svn_gh_migration/Blog_TheGreatMigration_car.jpg' /></a>
```

Comment:
Can't you just do a relative reference here?  That way you don't need to list a branch name and when you merge, it will be done.

Reply:
will that work to a different repo in the same org? If so, didn't know that.

Reply:
Are figures not allowed in this github repo?  Why not?  Could use git-lfs, no?

Reply:
Thats outside my pay grade ;) Am following pattern in place when I got here.

Reply:
Why make things more complex than they need to be?  @curfman and @maherou?

Reply:
Agree...it makes things more complex. Would be best if images and text just came as a sub-dir PR commit all in the same repo.

Reply:
Initial setup has been keeping images in a separate repo in order to keep repo size manageable. As we're taking a closer look at workflow for the site, we can reconsider pros/cons; let's discuss sometime.

Reply:
@curfman, I guess having the images stored in another repo on the 'master' branch is fine.  It is just that the 'master' branch of this repo should always be in a state where it can be published.  So all modifications to this github repo should be done on branches and reviewed with PRs.

I can create a PR to modify:

* https://github.com/betterscientificsoftware/betterscientificsoftware.github.io/blob/master/HowToContribute.md

to use PRs and forks correctly.  Okay?

Reply:
So, I would like to vote for chaning practice of separating images and content. Its two PRs to two sep. repos *and* where image link URLs need to then be modified as the PRs are marged. Its extra work and for what advantage? I think it would be better to have images and markdown files in same folder. We can employ LFS if GitHub bandwidth or storage limits become an issue. Do we know if BSSw.io is approaching those limits yet?

Reply:
The only concern with changing to use LFS is if GitHub will automatically render images when they are stored with LFS.  A simple experiment with some scratch github repo would tell us that.  But if that works, then there is no good reason to split out images into a separate github repo.

Reply:
So, I tried this using Preview tab to some LFS'd images in VisIt's repo. It works, for PNG images anyways.

Reply:
@bartlettroscoe, Thanks for considering workflow for the site; determining more rigorous processes is important as the site grows.  I suggest that you write a model for the fork/PR model that you have in mind, and then we can discuss with the editorial team and our partners at Sandbox and Parallactic, to ensure that the approach encompasses the full range of functionality needed.

Reply:
@markcmiller86, Your points about logistics for images make sense from the point of view of a content contributor; let's discuss with our partners at Sandbox and Parallactic to make sure that we also consider perspectives for image treatment and import of content to the front-end; I will follow up with them and we can discuss sometime.  For the immediate need for this blog article, please use the current image repo.  Thanks.